### ISSUE TITLE
LIS 4.0.11 not working with CentOS 7.2

### ISSUE THREAD

**Issue by NTICompass** [LIS/lis-next#43]
I just upgraded my CentOS VM to 7.2 (just by using `yum upgrade`).  This upgraded the kernel to `3.10.0-327.3.1.el7`.  After a reboot, I was unable to use my mouse and keyboard from a "Virtual Machine Connection" window.  Luckily, I was still able to use PuTTY to SSH into it.

In `dmesg` I saw the following:

```
[   30.880959] hyperv_keyboard: module verification failed: signature and/or required key missing - tainting kernel
[   30.880990] hyperv_keyboard: disagrees about version of symbol vmbus_driver_unregister
[   30.880992] hyperv_keyboard: Unknown symbol vmbus_driver_unregister (err -22)
[   30.881000] hyperv_keyboard: disagrees about version of symbol vmbus_sendpacket
[   30.881001] hyperv_keyboard: Unknown symbol vmbus_sendpacket (err -22)
[   30.881057] hyperv_keyboard: disagrees about version of symbol vmbus_close
[   30.881059] hyperv_keyboard: Unknown symbol vmbus_close (err -22)
[   30.881069] hyperv_keyboard: disagrees about version of symbol vmbus_recvpacket_raw
[   30.881070] hyperv_keyboard: Unknown symbol vmbus_recvpacket_raw (err -22)
[   30.881076] hyperv_keyboard: disagrees about version of symbol vmbus_open
[   30.881077] hyperv_keyboard: Unknown symbol vmbus_open (err -22)
[   30.881089] hyperv_keyboard: disagrees about version of symbol __vmbus_driver_register
[   30.881090] hyperv_keyboard: Unknown symbol __vmbus_driver_register (err -22)
[   30.903003] hv_balloon: disagrees about version of symbol vmbus_sendpacket
[   30.903048] hv_balloon: Unknown symbol vmbus_sendpacket (err -22)
[   30.903055] hv_balloon: disagrees about version of symbol vmbus_close
[   30.903056] hv_balloon: Unknown symbol vmbus_close (err -22)
[   30.903061] hv_balloon: disagrees about version of symbol vmbus_recvpacket
[   30.903062] hv_balloon: Unknown symbol vmbus_recvpacket (err -22)
[   30.903081] hv_balloon: disagrees about version of symbol vmbus_open
[   30.903082] hv_balloon: Unknown symbol vmbus_open (err -22)
[   30.903100] hv_balloon: disagrees about version of symbol __vmbus_driver_register
[   30.903101] hv_balloon: Unknown symbol __vmbus_driver_register (err -22)
[   30.903613] hv_balloon: disagrees about version of symbol vmbus_sendpacket
[   30.903620] hv_balloon: Unknown symbol vmbus_sendpacket (err -22)
[   30.903625] hv_balloon: disagrees about version of symbol vmbus_close
[   30.903626] hv_balloon: Unknown symbol vmbus_close (err -22)
[   30.903631] hv_balloon: disagrees about version of symbol vmbus_recvpacket
[   30.903632] hv_balloon: Unknown symbol vmbus_recvpacket (err -22)
[   30.903651] hv_balloon: disagrees about version of symbol vmbus_open
[   30.903652] hv_balloon: Unknown symbol vmbus_open (err -22)
[   30.903670] hv_balloon: disagrees about version of symbol __vmbus_driver_register
[   30.903671] hv_balloon: Unknown symbol __vmbus_driver_register (err -22)
[   30.949023] hid_hyperv: disagrees about version of symbol vmbus_driver_unregister
[   30.949029] hid_hyperv: Unknown symbol vmbus_driver_unregister (err -22)
[   30.949036] hid_hyperv: disagrees about version of symbol vmbus_sendpacket
[   30.949037] hid_hyperv: Unknown symbol vmbus_sendpacket (err -22)
[   30.949042] hid_hyperv: disagrees about version of symbol vmbus_close
[   30.949043] hid_hyperv: Unknown symbol vmbus_close (err -22)
[   30.949096] hid_hyperv: disagrees about version of symbol vmbus_recvpacket_raw
[   30.949098] hid_hyperv: Unknown symbol vmbus_recvpacket_raw (err -22)
[   30.949102] hid_hyperv: disagrees about version of symbol vmbus_open
[   30.949103] hid_hyperv: Unknown symbol vmbus_open (err -22)
[   30.949111] hid_hyperv: disagrees about version of symbol __vmbus_driver_register
[   30.949112] hid_hyperv: Unknown symbol __vmbus_driver_register (err -22)
[   30.949752] hyperv_fb: disagrees about version of symbol vmbus_driver_unregister
[   30.949757] hyperv_fb: Unknown symbol vmbus_driver_unregister (err -22)
[   30.949769] hyperv_fb: disagrees about version of symbol vmbus_sendpacket
[   30.949770] hyperv_fb: Unknown symbol vmbus_sendpacket (err -22)
[   30.949779] hyperv_fb: disagrees about version of symbol vmbus_close
[   30.949780] hyperv_fb: Unknown symbol vmbus_close (err -22)
[   30.949784] hyperv_fb: disagrees about version of symbol vmbus_recvpacket
[   30.949785] hyperv_fb: Unknown symbol vmbus_recvpacket (err -22)
[   30.949797] hyperv_fb: disagrees about version of symbol vmbus_open
[   30.949798] hyperv_fb: Unknown symbol vmbus_open (err -22)
[   30.949821] hyperv_fb: disagrees about version of symbol __vmbus_driver_register
[   30.949822] hyperv_fb: Unknown symbol __vmbus_driver_register (err -22)
```

**Comment by shirgall** [comment]
It is important to uninstall LIS before performing a kernel upgrade. Sometimes a kernel upgrade will reinstall LIS daemons or conflict with the LIS RPMs. AFter the kernel upgrade, reinstalling LIS will unload the distro versions of these packages and load the LIS ones in their place.

**Comment by NTICompass** [comment]
I didn't uninstall LIS *before*, but I did uninstall and then reinstall it afterwards.  That unfortunately didn't seem to help.

I think that something changed in the new kernel that needs to be addressed in the code.  I think RHEL72 (3.10.0-327) may need to be added to the LIS ISO.

**Comment by shirgall** [comment]
Sorry, I missed that you were talking about 7.2. Yes, getting 7.2 support into lis-next is currently in progress and is not complete.

**Comment by NTICompass** [comment]
Ok, cool.  I could find any info when I was searching, so I figured I'd post a bug.

**Comment by MHT980** [comment]
i have the exact same problem. 
kernel: hyperv_keyboard: disagrees about version of symbol vmbus_driver_unregister
CentOS Linux release 7.2.1511 (Core), kernel 3.10.0-327.3.1.el7.x86_64
It works with kernel 3.10.0-229.20.1.el7.x86_64
this only occurs with HyperV generation 2 virtual machine.

in generation 1 with same kernel and distro it works fine.

**Comment by NTICompass** [comment]
We'll just need to wait until LIS is updated for RHEL/CentOS 7.2

**Comment by marcinbojko** [comment]
I am really tired fighting with LIS modules every time I do yum-update. Instead of writing excuses everytime, just put the code in Linux kernel, as the others. Developement stopped on 3.1 and thats all.

**Comment by shirgall** [comment]
All of LIS gets submitted and accepted upstream eventually, and we do work with the distributions to get our work pulled in from there. The only reason for the separate package is for the timely availability of new features and fixes that have not worked their way through the upstream process yet. The "3.1" version number is an artifact of upstream inclusion and does not reflect the capabilities of the underlying Hyper-V drivers that have been included.

**Comment by marcinbojko** [comment]
Thanks Joshua. You convinced me to use distribution provided modules.
However, everything what Hyper-V is showing right now is just 'Detected' not a version number.

**Comment by shirgall** [comment]
The 7.2 kernel has a lot of support built in. We don't carry separate version numbers because what backports Red Hat takes is not a linear mapping to releases here. Bug fixes are easy, features are hard, and it all has to mesh with their release cycles. The work we are putting into LIS here will add 7.2 to the roster of supported versions, but it is also going to add more features that are not yet fully through the upstream process and backport process.

**Comment by shirgall** [comment]
This is now fixed with the release of LIS 4.1: https://www.microsoft.com/en-us/download/details.aspx?id=51612
